src/main/java/org/mariadb/jdbc/MariaDbClientPreparedStatement.java

```diff
@@ -160,10 +160,35 @@ public boolean execute() throws SQLException {
      *                               <code>ResultSet</code> object
      */
     public int executeUpdate() throws SQLException {
-        return executeUpdate(query);
+        if (executeInternal()) {
+            return 0;
+        }
+        return getUpdateCount();
     }
 
 
+    protected boolean executeInternal() throws SQLException {
+        executing = true;
+        QueryException exception = null;
+        lock.lock();
+        try {
+            executeQueryProlog();
+            batchResultSet = null;
+            Deque<ParameterHolder[]> tmpParam = new ArrayDeque<>(1);
```

Comment:
FYI, the smallest an ArrayDeque can allocate to is 8, and it must be a 2^x size.  It silently is just upsizing to that minimum size.  Not sure if you want to use a different structure for this now knowing that?  Maybe a Collections.singletonList?